msgpack-core/src/main/java/org/msgpack/core/MessagePacker.java

```diff
@@ -67,21 +67,65 @@
 import static org.msgpack.core.Preconditions.checkNotNull;
 
 /**
- * Writer of message packed data.
- * <p/>
+ * MessagePack serializer that converts objects into binary.
+ * You can use factory methods of {@link MessagePack} class or {@link MessagePack.PackerConfig} class to create
+ * an instance.
  * <p>
- * MessagePacker provides packXXX methods for writing values in the message pack format.
- * To write raw string or binary data, first use packRawStringHeader or packBinaryHeader to specify the data length,
- * then call writePayload(...) method.
- * </p>
- * <p/>
+ * This class provides following primitive methods to write MessagePack values. These primitive methods write
+ * short bytes (1 to 7 bytes) to the internal buffer at once. There are also some complex methods for convenience.
  * <p>
- * MessagePacker class has no guarantee to produce the correct message-pack format data if it is not used correctly:
- * packXXX methods of primitive values always produce the correct format, but
- * packXXXHeader (e.g. array, map, ext) must be followed by correct number of array/map/ext type values.
- * packRawStringHeader(length) and packBinaryHeader(length) must be followed by writePayload( ... length) to supply
- * the binary data of the specified length in the header.
- * </p>
+ * Primitive methods:
+ *
+ * <table>
+ *   <tr><th>Java type</th><th>Packer method</th><th>MessagePack type</th></tr>
+ *   <tr><td>null</td><td>{@link #packNil()}</td><td>Nil</td></tr>
+ *   <tr><td>boolean</td><td>{@link #packBoolean(boolean)}</td><td>Boolean</td></tr>
+ *   <tr><td>byte</td><td>{@link #packByte(byte)}</td><td>Integer</td></tr>
+ *   <tr><td>short</td><td>{@link #packShort(short)}</td><td>Integer</td></tr>
+ *   <tr><td>int</td><td>{@link #packInt(int)}</td><td>Integer</td></tr>
+ *   <tr><td>long</td><td>{@link #packLong(long)}</td><td>Integer</td></tr>
+ *   <tr><td>BigInteger</td><td>{@link #packBigInteger(BigInteger)}</td><td>Integer</td></tr>
+ *   <tr><td>float</td><td>{@link #packFloat(float)}</td><td>Float</td></tr>
+ *   <tr><td>double</td><td>{@link #packDouble(double)}</td><td>Float</td></tr>
+ *   <tr><td>byte[]</td><td>{@link #packBinaryHeader(int)}</td><td>Binary</td></tr>
+ *   <tr><td>String</td><td>{@link #packRawStringHeader(int)}</td><td>String</td></tr>
+ *   <tr><td>List</td><td>{@link #packArrayHeader(int)}</td><td>Array</td></tr>
+ *   <tr><td>Map</td><td>{@link #packMapHeader(int)}</td><td>Map</td></tr>
+ *   <tr><td>custom user type</td><td>{@link #packExtensionTypeHeader(byte, int)}</td><td>Extension</td></tr>
+ * </table>
+ *
+ * <p>
+ * Complex methods:
```

Comment:
The word complex has a little bit negative. We can say this as convenience, utility or helper method.